src/Hl7.Fhir.Core/Rest/TypedEntryResponseToBundle.cs

```diff
@@ -56,11 +56,11 @@ public static Bundle.EntryComponent ToBundleEntry(this TypedEntryResponse entry,
                 }
                 else
                 {
-                    if(entry.TypedElement != null)
+                    if (entry.TypedElement != null)
                     {
                         try
                         {
-                            result.Resource = new BaseFhirParser(parserSettings).Parse<Resource>(entry.TypedElement);
+                            result.Resource = new BaseFhirParser(parserSettings).Parse<Resource>(entry.TypedElement, inspector);
```

Comment:
I think we should introduce a protected `Parse()` method on the base fhir client - since that now already has the inspector, so we don't need to pass it around here. Also, by centralizing the Parse() in the base, we can more easily swap out different parsers/serializers (thinking ahead to using the other parsers here).

Reply:
Should we not make something injectable? Like an extra parameter in the `BaseFhirClient`? I investigated this a bit and we need the following:
- parsing:
  - json string => Resource
  - xml string => Resource
- serializing:
  - Resource => json string => byte[]
  - Resource => xml string => byte[]

Reply:
Yes, I think so too.

Reply:
This will be an extra task (and another PR). It is on the todo list